include/net/net_ip.h

```diff
@@ -157,6 +157,12 @@ struct sockaddr_ll_ptr {
 	u8_t        *sll_addr;    /* Physical-layer address  */
 };
 
+struct sockaddr_can_ptr {
```

Comment:
Any hint to add to commit message and the PR description whether this follows some well-known CAN Socket API. E.g., Linux'? If no, the suggestion is to follow up Linux, up to providing the sample which shares the code and works the same way on Linux vs Zephyr.

Reply:
done

Reply:
> From user point of view, the BSD socket CAN support
works same way as in Linux.

;-) Well, I guess it's better than nothing ;-).